snuba/consumers/consumer.py

```diff
@@ -179,13 +179,37 @@ def join(self, timeout: Optional[float] = None) -> None:
         )
 
 
+class MockReplacementBatchWriter(ProcessingStep[ReplacementBatch]):
+    """
+    Fake replacement writer used to run consumer load tests.
+    Contrarily to the mock writer. THis one does not introduce a delay.
+    This is because replacement are rare enough that adding the
```

Comment:
I'm not sure I understand why there's no delay here. Replacements take non-zero time to process, and isn't the intent here to test what happens when we have a lot of replacements?

Reply:
the main consumer only produces a replacement message on the replacement topic. 
The consumer that processes the replacement is another one and it does not use this code path.
This code is the one that takes the replacement message on the events topic, and sends it on the replacement topic.

Reply:
That doesn't really answer my question. Sending the replacement on takes a non-zero amount of time. Why would we test with no delay?

You also  say "replacement are rare enough..." but isn't the point of this code to test what happens when replacements are not rare?

Reply:
>  isn't the point of this code to test what happens when replacements are not rare?

Not really, this is not about testing replacements. This tests the main consumer on production load (where replacement messages are very rare). The main consumer needs a replacement writer where to produce replacement messages. If we do not provide a mock, the consumer we are testing will produce replacement on the real replacements topic so the test will not be idempotent. 

> Why would we test with no delay?

Because replacements messages are exceedingly rare for the consumer, we are going to test in production like conditions. It is really unlikely that ignoring that delay could impact the overall results.

Reply:
> we are going to test in production like conditions.

We are going to test the main errors consumer under production conditions. 
Compared with the amount of messages it receives containing events to write, the replacements messages are a tiny fraction. 
Each one of the replacement messages is just produced on a kafka topic basically with no processing.
Producing on Kafka is largely an asynchronous operation. When we close the batch we just ensure that all the callbacks have been executed. 
Since the replacements are scattered along the batch and they are a tiny fraction, by the time we close a batch all callbacks have generally been executed already. The callbacks do almost nothing. So The impact of this process on the consumer workload is very likely negligible.